### ISSUE TITLE
conv2 input

### ISSUE THREAD

**Issue by zhangajie** [aimerykong/deepImageAestheticsAnalysis#14]
Hi,thank you very much for your code, I have learned a lot from it.  I have a small problem which is not very clear, I hope you can take the time to help me out.  I visualize the trained caffemodel, and  I found that the num_input of the conv2 is 48, while the num_output of cov1 is 96. What makes it happen? It will be great if you can help me ,thank you!

**Comment by aimerykong** [comment]
Hi,

Thanks for your interest.

I believe that is because of the nature of Alexnet (
https://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf
)

In Alexnet, after conv1 (including pooling and normalization layers), all
the feature maps are split into two parts, each of them have 48 channels.
Then there are two columns of network computing over the two parts
respectively. The reason why Alexnet adopts this setup up is that GPU in
old days were not big enough to hold large amount of data.

Regards,
Shu

**Comment by zhangajie** [comment]
Hi, Shu
Thank you for you help! Since I haven read the paper, I am still a little confused because  the 'initModel.prototxt' has not the content about it. If I want to train it myself,  what should I do to split it, adding split layers?  Thanks again for your help!

**Comment by aimerykong** [comment]
Hi,

If you are using Caffe, you don't need to to anything other than providing
the prototxt file to define the architecture and the pre-trained model
(over ImageNet). In the convolution layer of caffe, there is one attribute
called "group" (as highlighted by red color below) that enables splitting
into half. So once you are using the prototxt provided, you can directly
run to train/fine-tune.

Regards,
Shu


layer {
  name: "conv2"
  type: "Convolution"
  bottom: "norm1"
  top: "conv2"
  param {
    name: "conv2_w"
    lr_mult: 0
    decay_mult: 1
  }
  param {
    name: "conv2_b"
    lr_mult: 0
    decay_mult: 0
  }
  convolution_param {
    num_output: 256
    pad: 2
    kernel_size: 5
    *group: 2*
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 1
    }
  }
}


Regards,
Shu

**Comment by zhangajie** [comment]
Thank you for your reply, I get it now. Thank you very much!

**Comment by ayantian** [comment]
Hi, 
Thank you very much for sharing your code. One thing that puzzles me is that  the code  dose not provide the image list used in "mergedNetRank.prototxt". I wonder how you select pairs of images. I also read your paper, I noticed that you investigated two strategies for selecting pairs of images used in computing the ranking loss. One way is to select pairs of images with a relatively large difference in their average aesthetic scores. The other way is sample image pairs that have been scored by the same individual. I want to know how this relatively large difference is defined, whether there is a need to set a specific threshold for the difference in aesthetic scores, and how many pairs of images you used for train.
 I am very looking forward to your help. 
Best wishes!
Chen

**Comment by aimerykong** [comment]
Hi,

Thanks for your interest. I didn't provide the list of image pairs because
the txt file that stores the pairs is >200MB, meaning millions pairs are
sampled for testing different strategies.

As for sampling pairs with large margin, we chose those image pairs with at
least 2 score difference, e.g. 3 vs. 5, 4 vs. 2, 3 vs. 1.

Hope this helps.

Regards,
Shu
